tests/Tags/TagsTest.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Qless\Tests\Tags;
+
+use Qless\Tests\QlessTestCase;
+
+class TagsTest extends QlessTestCase
+{
+    public function testPutWithEmptyTags(): void
+    {
+        $queue = $this->client->queues['test-queue'];
+        $queue->put('Foo', [], 'jid-11', 0, 0, 0, ['', '  ', ' ', 'tag-test1', 'tag-test2']);
+
+        self::assertEquals(['tag-test1', 'tag-test2'], $this->client->jobs['jid-11']->tags);
+    }
+
+    public function testPutAndTagWithEmptyTags(): void
+    {
+        $queue = $this->client->queues['test-queue'];
+
+        $queue->put('Foo', [], 'jid-12', 0, 0);
+        $queue->pop()->tag('', ' ', '   ', 'tag-test3', 'tag-test4');
+
+        self::assertEquals(['tag-test3', 'tag-test4'], $this->client->jobs['jid-12']->tags);
+    }
+
+    public function testPutWithEmptyTagsAndTagWithEmptyTags(): void
+    {
+        $queue = $this->client->queues['test-queue'];
+
+        $queue->put('Foo', [], 'jid-13', 0, 0, 0, ['', 'tag-test5', 'tag-test6', ' ']);
+        $queue->pop()->tag('', ' ', '   ', ' tag-test8');
+
+        self::assertEquals(['tag-test5', 'tag-test6', ' tag-test8'], $this->client->jobs['jid-13']->tags);
```

Comment:
Controversial point with spaces in tags. I think they should be trimmed too. ` test-tag` and `test-tag` means the same tag.

Reply:
This will already be the transformation of the tag. In my opinion, it is better to make these changes in a separate ticket.